public_html/wp-content/plugins/wordcamp-reports/classes/report/class-wordcamp-speaker-feedback.php

```diff
@@ -0,0 +1,393 @@
+	}
+
+	/**
+	 * Query and parse the data for the report.
+	 *
+	 * @return array
+	 */
+	public function get_data() {
+		// Bail if there are errors.
+		if ( ! empty( $this->error->get_error_messages() ) ) {
+			return array();
+		}
+
+		// Maybe use cached data.
+		$data = $this->maybe_get_cached_data();
+		if ( is_array( $data ) ) {
+			return $data;
+		}
+
+		// This script is a bit of a memory hog.
+		ini_set( 'memory_limit', '512M' ); // phpcs:ignore WordPress.PHP.IniSet.memory_limit_Blacklisted
+
+		$data      = array();
+		$wordcamps = $this->get_wordcamps();
+
+		foreach ( $wordcamps as $wordcamp ) {
+			$blog_id = get_wordcamp_site_id( $wordcamp );
+			if ( ! $blog_id ) {
+				continue;
+			}
+
+			$event_data = array(
+				'id'         => $wordcamp->ID,
+				'name'       => get_wordcamp_name( $blog_id ),
+				'url'        => get_post_meta( $wordcamp->ID, 'URL', true ),
+				'start_date' => get_post_meta( $wordcamp->ID, 'Start Date (YYYY-mm-dd)', true ),
+			);
+
+			switch_to_blog( $blog_id );
+
+			if ( should_generate_stats() ) {
+				$blog_data = gather_data();
+				$stats     = generate_stats( $blog_data );
+
+				$data[] = array_merge( $event_data, $stats );
+			}
+
+			restore_current_blog();
+		}
+
+		$this->maybe_cache_data( $data );
+
+		return $data;
+	}
+
+	/**
+	 * Compile the report data into results.
+	 *
+	 * @param array $data The data to compile.
+	 *
+	 * @return array
+	 */
+	public function compile_report_data( array $data ) {
+		return $data;
+	}
+
+	/**
+	 * Load the necessary files from the SFT plugin, since it's not activated on Central.
+	 *
+	 * @return void
+	 */
+	protected function load_sft_dependencies() {
+		$path = WP_PLUGIN_DIR . '/wordcamp-speaker-feedback/includes/';
+		require_once $path . 'class-feedback.php';
+		require_once $path . 'comment.php';
+		require_once $path . 'post.php';
+		require_once $path . 'view.php';
+		require_once $path . 'spam.php';
+		require_once $path . 'stats.php';
+	}
+
+	/**
+	 * Get data from a WordCamp Details report and filter it.
+	 *
+	 * @return array|null
+	 */
+	protected function get_wordcamps() {
+		$post_args = array(
+			'post_type'           => WCPT_POST_TYPE_ID,
+			'post_status'         => WordCamp_Loader::get_public_post_statuses(),
+			'posts_per_page'      => 9999,
+			'nopaging'            => true,
+			'no_found_rows'       => false,
+			'ignore_sticky_posts' => true,
+			'orderby'             => 'id',
+			'order'               => 'ASC',
+		);
+
+		if ( $this->range instanceof Date_Range ) {
+			// This replaces the default meta query.
+			$post_args['meta_query'] = array(
+				array(
+					'key'     => 'Start Date (YYYY-mm-dd)',
+					'value'   => array( $this->range->start->getTimestamp(), $this->range->end->getTimestamp() ),
+					'compare' => 'BETWEEN',
+					'type'    => 'NUMERIC',
+				),
+			);
+			$post_args['orderby'] = 'meta_value_num title';
+		}
+
+		if ( $this->wordcamp_id ) {
+			$post_args['post__in'] = $this->wordcamp_id;
```

Comment:
I also got an `array_map` error, I think because this is expected to be an array value.

Reply:
Fixed in f123b088